lib/src/utf8.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library utf8;
+
+import 'dart:convert';
+import 'dart:ffi';
+import 'dart:typed_data';
+
+/// [Utf8] implements conversion between Dart strings and null-termianted
+/// Utf8-encoded "char*" strings in C.
+class Utf8 extends Struct<Utf8> {
+  static String fromUtf8(Pointer<Utf8> str) {
```

Comment:
Consider naming parameter `string` (avoid abbreviations in general).

Also document of the function, e.g.:

/// Creates a [String] containing the characters UTF-8 encoded in [string].
///
/// The [string] must be a zero-terminated byte sequence of valid
/// UTF-8 encodings of Unicode code points.. 
/// It may also contain UTF-8 encodings of unpaired surrogate code points,
/// which is not otherwise valid UTF-8, but which may be created
/// when encoding a Dart string containing an unpaired surrogate.
///
/// Returns a Dart string containing the decoded code points.

(If that is actually correct, but it suggests what to test).

Reply:
Done.